src/libAtomVM/spinlock.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of AtomVM.
+ *
+ * Copyright 2022 Paul Guyot <pguyot@kallisys.net>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+ */
+
+#ifndef _SPINLOCK_H_
+#define _SPINLOCK_H_
+
+#ifndef AVM_NO_SMP
+#ifndef __cplusplus
+
+#include <stdatomic.h>
+
+#define spinlock_def(name) bool _Atomic name
+#define spinlock_init(name) name = false
+
+#define SPIN_LOCK(name) spinlock_lock(&name)
+#define SPIN_UNLOCK(name) spinlock_unlock(&name)
+
+static inline void spinlock_lock(bool _Atomic *lock)
+{
+    while (true) {
+        bool expected = false;
+        if (atomic_compare_exchange_weak_explicit(lock, &expected, true, memory_order_acquire, memory_order_relaxed))
```

Comment:
for consistency reasons I wouldn't use single line if statement, I'd rather use:
```
if (...) {
    break;
}
```
style.

Reply:
Noted. This apparently is an
option of clang-format 15 (InsertBraces). Should we upgrade it?